### PR TITLE
Menu tweaks

### PR DESCRIPTION
**Pull request by temx**
- Add FOV slider
- Mouse support for sliders
- Mouse support for scrollbars

### CODE REVIEW

File: Quake/menu.c [hunk 1]
```diff
@@ -1157,20 +1185,37 @@ void M_Menu_Options_f (void)
...
+		else if (f > 130)
+			f = 130;
+		Cvar_SetValue ("fov", f);
+		break;
 	case OPT_SCALE: // console and menu scale
 		if (scr_relativescale.value)
 		{
 			l = 3.0f;
-			f = scr_relativescale.value + dir * .1;
+			if (mouse)
```
Comment (Novum) [inline Quake/menu.c:1215]: I think we should add a function for those to remove the repetition of this if/else pattern?

Comment (temx) [reply Quake/menu.c]: Yeah, it should be better. I've factored it out now.

**Comment by Novum** [comment]
Looks good, thanks. Still want to verify at home before merging.

**Comment by Novum** [comment]
So I tested this and what I didn't like is that apparently one can't drag the sliders? Kind of unexpected.

**Comment by temx** [comment]
I added dragging support, and also an option for autoload (and changed its default). If you don't want it it can be omitted.

**Comment by Novum** [comment]
One more request, I think the sliders should snap to the values accessible via keyboard?

**Comment by temx** [comment]
Screen size will now snap above 100. It's a bit strange but no reason not to be continuous in the < 100 range. The others are continuous things so I left them as-is (you could already set any value you wanted in the console). If you think any sliders should be discrete it's a matter of changing the corresponding 999 value to 0. Also keyboard adjustments always re-snap now.

**Comment by Novum** [comment]
Thanks, feels great now.

**Pull request merged**